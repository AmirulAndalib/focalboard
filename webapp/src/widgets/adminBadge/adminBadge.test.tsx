// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react'
import {render} from '@testing-library/react'
import '@testing-library/jest-dom'

import {wrapIntl} from '../../testUtils'

import AdminBadge from './adminBadge'

describe('widgets/adminBadge', () => {
    test('should match the snapshot on TeamAdmin', () => {
        const {container} = render(wrapIntl(<AdminBadge permissions={['manage_team']}/>))
        expect(container).toMatchSnapshot()
    })

    test('should match the snapshot on Admin', () => {
        const {container} = render(wrapIntl(<AdminBadge permissions={['manage_team', 'manage_system']}/>))
        expect(container).toMatchSnapshot()
    })

    test('should match the snapshot on hide', () => {
        const {container} = render(wrapIntl(<AdminBadge permissions={[]}/>))
        expect(container).toMatchInlineSnapshot('<div />')
    })
})
