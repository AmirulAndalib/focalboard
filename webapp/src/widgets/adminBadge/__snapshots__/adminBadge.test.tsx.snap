// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`widgets/guestBadge should match the snapshot on show 1`] = `
<div>
  <div
    class="AdminBadge"
  >
    <div
      class="AdminBadge__box"
    >
      Admin
    </div>
  </div>
</div>
`;
